### PR TITLE
SCRUM-2282: test scaffold

### PR DESCRIPTION
**Pull request by abecerra**
[Incomplete - WIP]
Just to get some feedback, I added an `isMounted` check in the dashboard to avoid a warning when runing the test, it does not affect its functionality

### CODE REVIEW

**Comment by adamgibs** [comment]
@abecerra Looks good to me. One thing is that I don't think dashboardPage.test.js needs to set 'DashboardTableSettings' in local storage.

**Comment by abecerra** [comment]
@adamgibs : I think now it is ready. I need to learn how to check for the mocked data

File: src/main/cliapp/src/containers/conditionRelationPage/__tests__/ConditionRelationTable.test.js [hunk 1]
```diff
@@ -18,19 +19,32 @@ describe("<ConditionRelationPage />", () => {
 	});
 
 	it("Renders without crashing", async () => {
-		const result = renderWithClient(<ConditionRelationPage />);
-		expect(result);
+		var result;
+		act(() => {
```
Comment (adamgibs) [inline src/main/cliapp/src/containers/conditionRelationPage/__tests__/ConditionRelationTable.test.js:23]: These `act()` functions get rid of that warning when running the test which is awesome. Only thing I would say is change is change the `var` to `let.`

File: src/main/cliapp/src/containers/dashboardPage/Dashboard.js [hunk 2]
```diff
@@ -54,31 +54,35 @@ export const Dashboard = () => {
...
 	useEffect(() => {
+		let isMounted = true;
 		const systemService = new SystemService();
 
 		let _tableData = {};
 
 		systemService.getSiteSummary().then((res) => {
-			for (const key in res.entity){
+			if (isMounted) {
+				for (const key in res.entity){
```
Comment (adamgibs) [inline src/main/cliapp/src/containers/dashboardPage/Dashboard.js:64]: It's odd that this is needed. I would think wrapping the `render()` functions with `act()` would make this not necessary. Odd,  but works for now and doesn't affect functionality.   ¯\_(ツ)_/¯

**Review by adamgibs — APPROVED** [review]

File: src/main/cliapp/src/containers/conditionRelationPage/__tests__/ConditionRelationTable.test.js (hunk 1 above)
Comment (abecerra) [reply src/main/cliapp/src/containers/conditionRelationPage/__tests__/ConditionRelationTable.test.js:23]: I'll change them in next pr

File: src/main/cliapp/src/containers/dashboardPage/Dashboard.js (hunk 2 above)
Comment (abecerra) [reply src/main/cliapp/src/containers/dashboardPage/Dashboard.js:64]: Maybe the isMounted can be removed now, I'll test it

**Pull request merged**